background-processor/src/lib.rs

```diff
@@ -0,0 +1,240 @@
+use lightning::chain;
+use lightning::chain::chaininterface::{BroadcasterInterface, FeeEstimator};
+use lightning::chain::keysinterface::{ChannelKeys, KeysInterface};
+use lightning::ln::channelmanager;
+use lightning::ln::channelmanager::ChannelManager;
+use lightning::{log_internal, log_error, log_trace};
+use lightning::util::logger::Logger;
+use lightning::util::ser::Writeable;
+use std::sync::Arc;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::thread;
+use std::time::{Duration, Instant};
+
+/// BackgroundProcesser runs a background thread that monitors whether the ChannelManager it's given
+/// on start has new updates to persist. If it does, then it writes said ChannelManager to
+/// disk/backups. Note that if persisting the manager results in an error, BackgroundProcesser will
+/// log the error but nothing else -- so users are responsible for noticing when their supplied
+/// implementer of `channelmanager::Persist` fails. Fortunately, no funds can be lost if
+/// ChannelManager persistence fails.
+pub struct BackgroundProcessor {
+	stop_thread: Arc<AtomicBool>,
+}
+
+// Re-export these modules for the logging macros.
+mod util {
+	pub(crate) mod logger {
+		pub(crate) use lightning::util::logger::Record;
+	}
+}
+
+#[cfg(not(test))]
+const CHAN_FRESHNESS_TIMER: u64 = 60;
+#[cfg(test)]
+const CHAN_FRESHNESS_TIMER: u64 = 1;
+
+impl BackgroundProcessor {
+	/// Start the background thread that checks whether new updates are available for the ChannelManager,
+	/// and writes it to disk if so.
+	/// We also call ChannelManager::timer_chan_freshness_every_min(), since we're already eating the
+	/// cost of running a thread, we may as well pack as many operations into it as make sense.
+	/// Marked as must_use because otherwise the result is dropped immediately, resulting in the thread
+	/// being terminated.
+	#[must_use]
+	pub fn start<ChanSigner: 'static + ChannelKeys + Writeable, M: 'static + chain::Watch<Keys=ChanSigner>, T: 'static + BroadcasterInterface, K: 'static + KeysInterface<ChanKeySigner=ChanSigner>, F: 'static + FeeEstimator, L: 'static + Logger>(persister: Arc<dyn channelmanager::Persist<ChanSigner, M, T, K, F, L>>, manager: Arc<ChannelManager<ChanSigner, Arc<M>, Arc<T>, Arc<K>, Arc<F>, Arc<L>>>, logger: Arc<L>) -> Self {
+		let stop_thread = Arc::new(AtomicBool::new(false));
+		let stop_thread_clone = stop_thread.clone();
+		// See ChannelManager::persistence_lock for more documentation on the persistence process.
+		thread::spawn(move || {
+			let mut current_time = Instant::now();
+			loop {
+				let updates_available = manager.persistence_notifier.block_until_needs_persist(Some(Duration::from_millis(100)));
+				if updates_available {
+					if let Err(e) = persister.persist_manager(manager.clone()) {
+						log_error!(logger, "Errored persisting manager: {}", e);
```

Comment:
Error is only logged and discarded ? I acknowledge it's documented but look a bit surprising. Maybe a node owner could be willingly to stop processing ?

BAAFU ?

Reply:
What's BAAFU?